rancher/distros-test-framework
# Node replacement  test

#### Proposed Changes ####
 


- Adding node replace tests and its needed dependencies

- Functionalities added:
```
. SCP
. DeleteNodes
. Update kubeconfig files 
. Fetch server token 
. Get node name By IP
```


 - refactor aws code to return private ips as well, since our install scripts now requires that 


- separated upgrade stuff from each other



 - AUX.GO 79 -  Log statement removing cause stdout is already returned at this point so no need to also log it + sometimes it can contains sensitive data here 


-  increase to the highest normally supported line lenght ( i would not open another new PR just for this one single number change  ) 


 







#### Types of Changes ####


- Code design not too generic on parsing files and calling it, due to the complex logic of this test and funcs , but i will talk in a standup, explaning better





#### Testing ####


Checklist:
1. If your PR changes anything on or related to Jenkins, run it pointing to your branch to make sure it's okay.

- yes , jenkins created and will be testing before merge

2. Verify code lint; we should not have errors.

- yes 

3. Update the documentation if needed.

- yes 

4. Update makefile and docker run if adding new tests.

- yes 

5. Run your tests at least 4 times with all configurations needed and possible.

- ran 299929 times actually lOL

6. If needed test with different os types.

- tested in rhel , sles and ubuntu 


#### Linked Issues ####



#### Further Comments ####
